engineapi/engine.go

```diff
@@ -228,3 +228,16 @@ func (e *Engine) BackupRestoreIncrementally(backupTarget, backupName, backupVolu
 
 	return nil
 }
+
+func (e *Engine) BackupRestoreStatus() ([]types.RestoreStatus, error) {
+	args := []string{"backup", "restore-status"}
+	output, err := e.ExecuteEngineBinary(args...)
+	if err != nil {
+		return nil, err
+	}
+	restoreStatusList := make([]types.RestoreStatus, 0)
```

Comment:
In the old PR, I've commented it's better to make the result a `map` with replica name as the key. I still think it's better.

Reply:
Do you want the engine to return the map or do you want the manager to create a map out of a slice?